File: Backend/NootsWEB/Noots.Editor/Entities/Text/LetterDiff.cs


namespace Noots.Editor.Entities.Text;

public class LetterDiff
{
    public DiffOperation Operation { set; get; }

    public string Letter { set; get; }
}

public enum DiffOperation
{
    DELETE = -1,
    SAME = 0,
    ADD = 1,
}

File: Client/src/app/content/notes/full-note/content-editor/text/entities/diffs/letter-diff.ts
export class LetterDiff {
  public operation: DiffOperation;

  public str: string;

  public index: number;

  public shiftedIndex: number;

  constructor(operation: DiffOperation, str: string, index: number, shift: number) {
    this.operation = operation;
    this.str = str;
    this.index = index;
    this.shiftedIndex = shift;
  }
}

export enum DiffOperation {
  DELETE = -1,
  SAME = 0,
  ADD = 1,
}
